### PR TITLE
Helping swift to infer type of 'complex' expression when used as a pod

### PR DESCRIPTION
**Pull request by kirillyakimovich**
Without this compiler is dying with "Ambiguous use of 'map'"

### CODE REVIEW

**Comment by kzaher** [comment]
Interesting, what compiler version are you using? This works just fine with Xcode Version 9.2 (9C40b)

File: RxCocoa/Common/DelegateProxy.swift [hunk 1]
```diff
@@ -208,9 +208,9 @@
             #endif
             self._setForwardToDelegate(delegate, retainDelegate: retainDelegate)
 
-            let allUsedSelectors: [Selector] =
-                    self._sentMessageForSelector.values.filter { $0.hasObservers }.map { $0.selector } +
-                    self._methodInvokedForSelector.values.filter { $0.hasObservers }.map { $0.selector }
+            let sentSelectors: [Selector] = _sentMessageForSelector.values.filter { $0.hasObservers }.map { $0.selector }
```
Comment (kzaher) [inline RxCocoa/Common/DelegateProxy.swift]: Please use `self._sentMessageForSelector` and `self._methodInvokedForSelector`.

**Comment by kirillyakimovich** [comment]
Same Xcode as you: Version 9.2 (9C40b)
Trying to create minimal project when this will fail to provide some more info.

**Comment by kirillyakimovich** [comment]
Ok, i found the reason my specified swift version was 3.2. Sorry for bothering. Closing

**Pull request merged**